src/Sitewards/DBCompare/Worker/DBWorker.php

```diff
@@ -26,15 +28,22 @@ class DBWorker
     /**
      * @param string $sDBUsername
      * @param string $sDBPassword
-     * @param int $iItemWorkerId
+     * @param string $sItemWorkerId
+     * @param ItemWorkerFactory $oItemFactory
      */
-    public function __construct($sDBUsername, $sDBPassword, $iItemWorkerId)
+    public function __construct(
+        $sDBUsername,
+        $sDBPassword,
+        $sItemWorkerId,
+        ItemWorkerFactory $oItemFactory
+    )
     {
         $this->sDBUsername = $sDBUsername;
         $this->sDBPassword = $sDBPassword;
-        $this->iItemWorkerId = $iItemWorkerId;
+        $this->sItemWorkerId = $sItemWorkerId;
 
         $this->oConnection = DBConnectionFactory::getDatabaseConnection($sDBUsername, $sDBPassword);
```

Comment:
We could pass the DBConnection object instead of username and password thus simplifying this object